### PR TITLE
Fixing important issues with synchrotron extension

### PR DESCRIPTION
**Pull request by FilipO28555**
Done:
 - [x] Changed integration of first synchrotron function to not throw errors but deal with the inputs.
 - [x] In validate.py I added a check if normalized energy is <1 - if not that means that propably the electron recoil was enabled for the test 
 - [x] Electron recoil disabled for the .ci test
 - [x] Added docs and Copyright to change_parameters_randomly.py

Possible further work for the [next pull requests](https://github.com/ComputationalRadiationPhysics/picongpu/pull/4926) see [issue #4912](https://github.com/ComputationalRadiationPhysics/picongpu/issues/4912)

### CODE REVIEW

File: include/picongpu/simulation/stage/SynchrotronRadiation.hpp [hunk 1]
```diff
@@ -125,8 +127,13 @@ namespace picongpu::simulation::stage
...
                 {
-                    std::cout << "Caught exception when precomputing firstSynchrotronFunction at " << i
-                              << ". index: " << e.what() << std::endl;
+                    std::cout << "Caught exception when precomputing firstSynchrotronFunction at index " << i << ". "
+                              << e.what() << std::endl;
+                    std::cout << "zq: " << zq << ", xLeft: " << xLeft << ", xMiddle: " << xMiddle
+                              << ", xRight: " << xRight << std::endl;
+                    float_64 yLeft = std::cyl_bessel_k(5.0 / 3.0, xLeft);
+                    float_64 yMiddle = std::cyl_bessel_k(5.0 / 3.0, xMiddle);
+                    std::cout << "yLeft: " << yLeft << ", yMiddle: " << yMiddle << std::endl << std::endl;
```
Comment (PrometheusPi) [inline include/picongpu/simulation/stage/SynchrotronRadiation.hpp]: Here are you catching the (potential) previous errors: 
If such an error arises, does it make sense to continue with the simulation:
if yes: keep it as it is
if no: make `std::cout` and `std::cerr` and kill the simulation by throwing another error.

**Review by PrometheusPi — DISMISSED** [review]

**Review by PrometheusPi — APPROVED** [review]

File: include/picongpu/simulation/stage/SynchrotronRadiation.hpp [hunk 2]
```diff
@@ -113,188 +118,180 @@ namespace picongpu::simulation::stage
...
+    /** precomputed first and second synchrotron functions:
+     *  -> 2d grid of floats_64 -> tableValuesF1F2[zq][0/1] ; 0/1 = F1/F2
+     */
+    std::shared_ptr<GridBuffer<float_X, 2>> tableValuesF1F2;
+    //! used for table access
+    enum struct Accessor : uint32_t
+    {
+        f1 = 0u,
+        f2 = 1u
     };
```
Comment (PrometheusPi) [inline include/picongpu/simulation/stage/SynchrotronRadiation.hpp]: according to the error message of the CI, you are a missing a `;`after the class

**Review by PrometheusPi — CHANGES_REQUESTED** [review]

File: include/picongpu/simulation/stage/SynchrotronRadiation.hpp (hunk 2 above)
Comment (FilipO28555) [reply include/picongpu/simulation/stage/SynchrotronRadiation.hpp]: done

File: share/picongpu/tests/Synchrotron_plugin/lib/validate.py [hunk 3]
```diff
@@ -84,7 +84,15 @@ def main(dataPath):
...
     Energy_e = momentum_to_energy(e_p_y, const.m_e).max()  # convert momentum to energy
 
-    hist_data /= Energy_e  # go to delta space
+    hist_data /= Energy_e  # go to delta space.
+    # The momentum precision is 6 digits so the energy is 3 digits precision
+    if np.max(hist_data) > 1 + 1e-3:
+        raise SystemExit(
+            f"\nMax normalized energy of photons is {np.max(hist_data):.6f} but it should be less than 1 + 1e-3. \n"
+            "Is electron recoil on?\n"
+        )
```
Comment (PrometheusPi) [inline share/picongpu/tests/Synchrotron_plugin/lib/validate.py]: Do we still need this check after figuring out that full energy transfer is not possible? 
If not, let's go back to 1 (even if this is too high).

**Review by PrometheusPi — CHANGES_REQUESTED** [review]

**Comment by PrometheusPi** [comment]
FYI: We investigated the energy description of 0.1% and still do not understand it. For now, this pull request should be merged. To track the energy issue, I opened the issue #4941

**Review by PrometheusPi — APPROVED** [review]

**Pull request merged**